### PR TITLE
Defect/swagger load fails on doc change

### PR DESCRIPTION
**Pull request by mrjsawdk**
Fix the following issue:
- The odata provider's cache implementation is through Lazy<Func<SwaggerDocument>>. If an error occurs within the Func then the result will still be cached and will never be cleared. Replaced it with a standard swagger implementation which invokes the func if not in the cache and if it fails generation then no cache entry is added
- Reject access for most of v1 including ui login for rights holders

### CODE REVIEW

File: Presentation.Web/App_Start/SwaggerConfig.cs [hunk 1]
```diff
@@ -76,9 +76,8 @@ public static void Register()
 
                     c.ResolveConflictingActions(apiDescriptions => apiDescriptions.First());
 
-                    c.CustomProvider(defaultProvider =>
-                        new ODataSwaggerProvider(defaultProvider, c, GlobalConfiguration.Configuration).Configure(
-                            odataConfig => { odataConfig.EnableSwaggerRequestCaching(); }));
+                    //Do not enable the build in caching in the odata provider. It caches error responses which we dont want so we wrap it in a custom caching provider which bails out on errors
```
Comment (JacobMalling) [inline Presentation.Web/App_Start/SwaggerConfig.cs:79]: "Do not enable the **built-in** caching in..."

File: Presentation.Web/Swagger/FixNamingOfComplexQueryParametersFilter.cs [hunk 2]
```diff
@@ -15,9 +16,10 @@ public class FixNamingOfComplexQueryParametersFilter : IOperationFilter
 
         public void Apply(Operation operation, SchemaRegistry schemaRegistry, ApiDescription apiDescription)
         {
-            if (operation.parameters != null && MatchGET(apiDescription))
+            var parameters = operation.parameters?.ToList() ?? new List<Parameter>();
+            if (parameters.Any() && MatchGET(apiDescription))
             {
-                foreach (var parameter in operation.parameters.Where(IsNestedInComplexType).ToList())
+                foreach (var parameter in parameters.Where(IsNestedInComplexType).ToList())
```
Comment (JacobMalling) [inline Presentation.Web/Swagger/FixNamingOfComplexQueryParametersFilter.cs:22]: Er der ikke dobbelt ToList() nu?

**Review by JacobMalling — APPROVED** [review]

File: Presentation.Web/Controllers/API/ItSystemUsageController.cs [hunk 3]
```diff
@@ -13,7 +13,6 @@
 using Core.DomainServices;
 using Core.DomainServices.Authorization;
 using Core.DomainServices.Extensions;
-using Presentation.Web.Extensions;
 using Presentation.Web.Infrastructure.Attributes;
```
Comment (JacobMalling) [inline Presentation.Web/Controllers/API/ItSystemUsageController.cs:16]: Ingen deny på denne?

File: Presentation.Web/Controllers/OData/ItSystemUsagesController.cs [hunk 4]
```diff
@@ -6,7 +6,6 @@
 using System.Net;
 using Core.DomainModel.ItSystemUsage;
 using Core.DomainServices;
-using Core.DomainModel.Organization;
 using Core.DomainModel.ItSystem;
```
Comment (JacobMalling) [inline Presentation.Web/Controllers/OData/ItSystemUsagesController.cs:9]: hvad med deny på odata?

**Review by JacobMalling — APPROVED** [review]

File: Presentation.Web/Controllers/OData/ItSystemUsagesController.cs (hunk 4 above)
Comment (mrjsawdk) [reply Presentation.Web/Controllers/OData/ItSystemUsagesController.cs:9]: kig i baseentitycontroller ;-)

File: Presentation.Web/Controllers/API/ItSystemUsageController.cs (hunk 3 above)
Comment (mrjsawdk) [reply Presentation.Web/Controllers/API/ItSystemUsageController.cs:16]: @JacobMalling kig i genericapicontroller ;-)

**Pull request merged**